File: C++/Queue/queue.cpp
#include<iostream>

using namespace std;

#define size 20

class Queue{
	
	private:
		int* array;
		int front;
		int back;
		
	public:
		Queue(){
			array = new int[size];
			front = -1;
			back = -1;
		}
		
		void EnQueue(int val){
			if(back==size-1){
				cout<<"Queue overflow"<<endl;
				return;
			}
			back++;
			array[back] =  val;
			
			if(front == -1){
				front++;
			}
		}
		
		void DeQueue(){
			if(front==-1 || front>back){
				cout<<"ERROR: No elements in queue"<<endl;
				return;
			}
			
			front = (front + 1) % size;
			
		}
		
		int Peek(){
			if(front==-1 || front>back){
				cout<<"ERROR: No elements in queue"<<endl;
				return -1;
			}
			return array[front];
		}
		
		bool isEmpty(){
			if(front==-1 || front>back){
				return true;
			}
			return false;
		}
};


int main() {
    Queue q;
    q.EnQueue(5);
    q.EnQueue(10);
    q.EnQueue(15);
    q.DeQueue();
    cout << q.Peek() << endl;
    return 0;
}

File: Python/Queue/queue.py
class Queue:
    def __init__(self):
        self.items = []

    def isEmpty(self):
            return len(self.items) == 0
    
    def enQueue(self, val):
            self.items.append(val)

    def deQueue(self):
        if self.isEmpty():
            return None
        return self.items.pop(0)
    
    def Peek(self):
        return self.items[0] if not self.isEmpty() else None

    def Size(self):
        return len(self.items)

queue = Queue()

# enqueue some items
queue.enQueue(1)
queue.enQueue(2)
queue.enQueue(3)

# get the size of the queue
print("Queue size: ", queue.Size())

# get the front item of the queue
print("Front item: ", queue.Peek())

# dequeue an item from the queue
dequeued_item = queue.deQueue()
print("Dequeued item: ", dequeued_item)

# check if the queue is empty
print("Is queue empty? ", queue.isEmpty())